mayastor/src/grpc/pool_grpc.rs

```diff
@@ -0,0 +1,238 @@
+                        Err(e)
+                    }
+                }
+            }
+            Ok(lvol) => {
+                debug!("created lvol {}", lvol);
+                Ok(lvol)
+            }
+            Err(e) => Err(e),
+        }
+    })
+}
+
+/// destroy the replica on the given pool, returning OK if the replica was
+/// not found
+#[instrument(level = "debug", err)]
+pub async fn destroy_replica(args: DestroyReplicaRequest) -> GrpcResult<Null> {
+    rpc_call(async move {
+        match Bdev::lookup_by_name(&args.uuid) {
+            Some(b) => {
+                let lvol = Lvol::try_from(b)?;
+                lvol.destroy().await.map(|_r| Null {})
+            }
+            None => Ok(Null {}),
+        }
+    })
+}
+
+/// list all the replicas
+#[instrument(level = "debug", err)]
+pub fn list_replicas() -> GrpcResult<ListReplicasReply> {
+    let mut replicas = Vec::new();
+    if let Some(bdev) = Bdev::bdev_first() {
+        replicas = bdev
+            .into_iter()
+            .filter(|b| b.driver() == "lvol")
+            .map(|b| Replica::from(Lvol::try_from(b).unwrap()))
+            .collect::<Vec<_>>();
+    }
+
+    Ok(Response::new(ListReplicasReply {
+        replicas,
+    }))
+}
+
+/// shares the replica over nvmf -- replicas are always shared over nvmf if
+/// already shared returns OK.
+///
+/// There is no unshare RPC in mayastor_svc
+#[instrument(level = "debug", err)]
+pub async fn share_replica(
+    args: ShareReplicaRequest,
+) -> GrpcResult<ShareReplicaReply> {
+    rpc_call(async move {
+        if let Some(b) = Bdev::lookup_by_name(&args.uuid) {
+            let lvol = Lvol::try_from(b)?;
+
+            // if we are already shared return OK
+            if lvol.shared() == Some(Protocol::from(args.share)) {
+                return Ok(ShareReplicaReply {
+                    uri: lvol.share_uri().unwrap(),
+                });
+            }
+            match Protocol::from(args.share) {
+                Protocol::None => {
+                    lvol.unshare().await.map(|_| ShareReplicaReply {
+                        uri: format!("bdev:///{}", lvol.name()),
+                    })
+                }
+
+                Protocol::Nvmf => {
+                    lvol.share_nvmf().await.map(|_| ShareReplicaReply {
+                        uri: lvol.share_uri().unwrap(),
+                    })
+                }
+                Protocol::Iscsi | Protocol::Invalid => {
+                    Err(LvsError::LvolShare {
+                        source: CoreError::NotSupported {
+                            source: Errno::ENOSYS,
+                        },
+                        name: args.uuid,
+                    })
+                }
+            }
+        } else {
+            Err(LvsError::InvalidBdev {
+                source: NexusBdevError::BdevNotFound {
+                    name: args.uuid.clone(),
+                },
+                name: args.uuid,
+            })
+        }
+    })
```

Comment:
looks good except minor issues with logging. We used to log a debug message at the start of each grpc method and info message for all methods that change the state of mayastor when completed. That is handy for seeing operations that either hang or take too long easily in the log file. I suppose that the debug message at the beginning is replaced by instrument macro now so that is probably ok. But I'm missing the info message. Or are we sure that messages get logged for all replica and pool methods somewhere deeper in the modules that get called? Also the previous log messages were nice to users and speak their language  (about pools, replicas) as opposed using terms lvols and lvs that people not familiar with SPDK won't have idea what they mean.

Reply:
That's still there when the debug level is set, let me know what you are missing.

Reply:
I misread it the OK messages as what you are referring to, I'll have a good look :+1: 

As for the terms, the messages still speak of pools and replica's although I would argue those concepts are somehow more familiar to them unless they are familiar with zfs :)